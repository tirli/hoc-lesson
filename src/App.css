.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.listItemContainer {
  display: flex;
  flex-direction: row;
}

.posterWrapper, .poster {
  width: 80px;
  height: 100px;
}

.listItemContainer {
  border: 1px solid black;
}

.content {
  line-height: 100px;
}
.contentWrapper {
  margin-left: 10px;
}
