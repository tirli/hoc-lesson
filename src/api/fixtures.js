export default {
  "Search": [
    {
      "Title": "The Test",
      "Year": "2013",
      "imdbID": "tt2616114",
      "Type": "movie",
      "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjMzMDQwMzM2M15BMl5BanBnXkFtZTcwMzA1OTg1OQ@@._V1_SX300.jpg"
    },
    {
      "Title": "Sound Test for Blackmail",
      "Year": "1929",
      "imdbID": "tt0249159",
      "Type": "movie",
      "Poster": "N/A"
    }
  ],
  "totalResults": "618",
  "Response": "True"
}