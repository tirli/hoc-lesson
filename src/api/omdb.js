const baseUrl = 'http://www.omdbapi.com/?';

export function search(title) {
  return fetch(`${baseUrl}s=${title}&r=json`)
    .then((res) => res.json())
    .then((result) => result.Search.map((item) => ({
      ...item,
      Poster: item.Poster === 'N/A' ? null : item.Poster,
    })))
  ;
}

export function getById(id, plotType = 'short') {
  return fetch(`${baseUrl}i=${id}&r=json&plot=${plotType}`)
    .then((res) => res.json());
};
