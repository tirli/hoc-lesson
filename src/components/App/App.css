.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.listItemContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.posterWrapper, .poster {
  width: 80px;
  height: 100px;
}

.listItemContainer {
  border: 1px solid black;
}

.listItemContainer.active {
  background-color: rgba(200, 9, 209, 0.3);
}

.movieListHeader {
  height: 40px;
  font-size: 18px;
  color: purple;
  align-items: center;
  display: flex;
  justify-content: center;
}

.content {
  line-height: 100px;
}
.contentWrapper {
  margin-left: 10px;
}
