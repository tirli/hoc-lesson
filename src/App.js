import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import { search } from './api/omdb';

import MovieList from './components/MovieList';
import SearchField from './components/SearchField';
import FilmInfo from './components/FilmInfo';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
      movieId: null,
    };
  }

  onSearch = (value) => {
    search(value).then((result) => this.setState({ items: result }));
  }

  handleSelectedMovie = (id) => {
    this.setState({ movieId: id });
  }

  render() {
    const { items, movieId } = this.state;

    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <SearchField onSearch={this.onSearch} />
        </div>
        <MovieList movies={items} onSelect={this.handleSelectedMovie}/>
        <FilmInfo itemId={movieId} />
      </div>
    );
  }
}

export default App;
