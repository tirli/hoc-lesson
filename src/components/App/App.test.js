import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import ReactTestUtils from 'react-addons-test-utils'

import * as api from '../../api/omdb';

describe('App', () => {
  it('renders without crashing', () => {
    window.performance = window.Date;
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('shallow renders as expected without snapshot', () => {
    const app = shallow(<App />).html();
    expect(app).toBe([
      '<div class="App">',
        '<div class="App-header">',
        '<img src="test-file-stub" class="App-logo" alt="logo"/>',
        '<form><input value=\"\" name=\"search\" class=\"test-input\"/><button class=\"test-search-btn\"> Search </button></form>',
        '</div>',
      '</div>',
    ].join(''));
  });

  it('shallow renders as expected', () => {
    const app = shallow(<App />).html();
    expect(app).toMatchSnapshot();
  });

  it('call search api after onSearch', () => {
    api.search = jest.fn(() => Promise.resolve());
    const app = mount(<App />);
    app.instance().onSearch('kate');
    expect(api.search).toHaveBeenCalledTimes(1);
    expect(api.search).toHaveBeenCalledWith('kate');
  });

})


