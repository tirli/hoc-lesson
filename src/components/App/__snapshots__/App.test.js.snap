exports[`App shallow renders as expected 1`] = `"<div class=\"App\"><div class=\"App-header\"><img src=\"test-file-stub\" class=\"App-logo\" alt=\"logo\"/><form><input value=\"\" name=\"search\" class=\"test-input\"/><button class=\"test-search-btn\"> Search </button></form></div></div>"`;
